### ISSUE TITLE
Tool.Clear is not working properly

### ISSUE THREAD

**Issue by tar04** [dilidili/react-drawing-board#30]
Hey there! I have an issue. Here is it: whem im getting my operations stored on backend theyre pasting to the drawboard properly but when im clearing it i can still choose deleted elements. How can i fix that?

**Comment by tar04** [comment]
@dilidili

**Comment by tar04** [comment]
I guess i have nice idea how to fix this. You just have to change file SelectTool.tsx a bit like this:

Function onSelectMouseDown now finding operation to select inside array of all operations. But you need to find operation inside array which consist of operations after clear operation was added. Theres code sample: 


```
const lastClearIndex = operationListState.reduced.length - [...operationListState.reduced].reverse().findIndex(operation => operation.tool === Tool.Clear)

const paintedOperations = operationListState.reduced.slice(lastClearIndex);

let selectedItem: Operation | null = findSelectedItem(paintedOperations, pos, scale);
```

**Comment by mudasirpandith** [comment]
@tar04 I need help. Can i setup it in the nextjs or reactjs ?Please guide

**Comment by tar04** [comment]
> @tar04 I need help. Can i setup it in the nextjs or reactjs ?Please guide

You need help with the issue or to setup whole lib?

**Comment by tar04** [comment]
@mudasirpandith

**Comment by mudasirpandith** [comment]
@tar04  I want to use editor in my website. that`s setting lib

**Comment by tar04** [comment]
> @tar04 I want to use editor in my website. that`s setting lib

Have you tried instructions from repository?

**Comment by tar04** [comment]
Or you have some specific problems?

**Comment by mudasirpandith** [comment]
> Or you have some specific problems?

I tried to use  `const DrawingBoard = require('react-drawing-board');` , but i am not able to do it.Showing some conflicts and bunch of error while installing it

**Comment by tar04** [comment]
@mudasirpandith what about es6 imports?

**Comment by mudasirpandith** [comment]
> @mudasirpandith what about es6 imports?

It is not getting installed.

**Comment by tar04** [comment]
> > @mudasirpandith what about es6 imports?
> 
> It is not getting installed.

What you mean?) you have to install library and types for it if you’re using typescript

**Comment by mudasirpandith** [comment]
i was trying to use it in the React and using Js. 
is this correct way to use install ` npm install react-drawing-board`

**Comment by tar04** [comment]
> i was trying to use it in the React and using Js. is this correct way to use install ` npm install react-drawing-board`

Yes

**Comment by tar04** [comment]
And then you have to import it inside component you want to use it in

**Comment by mudasirpandith** [comment]
![Screenshot 2023-12-22 183730](https://github.com/dilidili/react-drawing-board/assets/78873815/4d6922ae-77b4-49e1-8f38-241316c900b0)
![image](https://github.com/dilidili/react-drawing-board/assets/78873815/d0522db2-6427-481e-a64e-871654812c3b)
These are the errors it is showing

**Comment by tar04** [comment]
> ![Screenshot 2023-12-22 183730](https://private-user-images.githubusercontent.com/78873815/292493503-4d6922ae-77b4-49e1-8f38-241316c900b0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMyNTExMTcsIm5iZiI6MTcwMzI1MDgxNywicGF0aCI6Ii83ODg3MzgxNS8yOTI0OTM1MDMtNGQ2OTIyYWUtNzdiNC00OWUxLThmMzgtMjQxMzE2YzkwMGIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIyVDEzMTMzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBhOTAyNjdmMDY4ODBmNjU0YzM1YzY4M2I1ZWVkZGY4OWQyNDMyYmE2Njc3YmJjMDY2ZWMxOTc3ODk5Y2YwOTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fQih2MM0z92My4titV5O5hJupSjmUjWYRk5ubZzaaw0) ![image](https://private-user-images.githubusercontent.com/78873815/292493591-d0522db2-6427-481e-a64e-871654812c3b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMyNTExMTcsIm5iZiI6MTcwMzI1MDgxNywicGF0aCI6Ii83ODg3MzgxNS8yOTI0OTM1OTEtZDA1MjJkYjItNjQyNy00ODFlLWE2NGUtODcxNjU0ODEyYzNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIyVDEzMTMzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVjNzE5OTk2MGQ0MmZkYzBiNzM4MDI4YTc4OWZmYjFkMDExZjc3MjFlYTFmYmFhY2Y3MjAyZmI1NzU0MjZkNDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1N5nOwZnQU6x_lN-llqYjvIdtkrKvy2RPwfNzi7zTP8) These are the errors it is showing

Did you opened project folder in console?

**Comment by mudasirpandith** [comment]
Yes,i am installing in the right directory. Other npm gets installed in that

**Comment by mudasirpandith** [comment]
@tar04

**Comment by tar04** [comment]
What do you mean “other npm”? @mudasirpandith

**Comment by mudasirpandith** [comment]
> What do you mean “other npm”? @mudasirpandith

i mean other packages get installed. Is there any way I can use this editor in my website?

**Comment by tar04** [comment]
@mudasirpandith You can use cdn server instead, just find the link with the lib and add script with it to your project

**Comment by mudasirpandith** [comment]
> @mudasirpandith You can use cdn server instead, just find the link with the lib and add script with it to your project

can you help me out for this? please

**Comment by tar04** [comment]
@mudasirpandith sure. I’ll be home in 25 mins

**Comment by tar04** [comment]
@mudasirpandith https://www.jsdelivr.com/package/npm/react-drawing-board

**Comment by tar04** [comment]
Copy the script, paste in html document and then you should try to find component in window object

**Comment by tar04** [comment]
Btw what ide do u use?

**Comment by mudasirpandith** [comment]
> Btw what ide do u use?

vscode

**Comment by tar04** [comment]
You can add the package in package.jam file by yourself and the ide should show you hint about uninstalled dependencies

**Comment by tar04** [comment]
@mudasirpandith

**Comment by mudasirpandith** [comment]
I am totally lost into this.
Can you please guide me step by step?

> @mudasirpandith

**Comment by tar04** [comment]
> I am totally lost into this. Can you please guide me step by step?
> 
> > @mudasirpandith

Yup. Do you hame smth like discord? JUst to text you there

**Comment by mudasirpandith** [comment]
yeah, discord username :mudasirpandith

> > I am totally lost into this. Can you please guide me step by step?
> > > @mudasirpandith
> 
> Yup. Do you hame smth like discord? JUst to text you there

**Comment by tar04** [comment]
@mudasirpandith added
